test/geometry/test_depth.py

```diff
@@ -329,3 +329,77 @@ def test_gradcheck(self, device, dtype):
             raise_exception=True,
             fast_mode=True,
         )
+
+
+class TestDepthFromDisparity:
+    def test_smoke(self, device, dtype):
+        disparity = torch.rand(1, 1, 3, 4, device=device, dtype=dtype)
+        baseline = torch.rand(1, device=device, dtype=dtype).item()
+        focal = torch.rand(1, device=device, dtype=dtype)
+
+        points3d = kornia.geometry.depth.depth_from_disparity(disparity, baseline, focal)
+        assert points3d.shape == (1, 1, 3, 4)
+
+    @pytest.mark.parametrize("batch_size", [2, 4, 5])
+    def test_shapes(self, batch_size, device, dtype):
+        disparity = torch.rand(batch_size, 1, 3, 4, device=device, dtype=dtype)
+        baseline = torch.rand(4, device=device, dtype=dtype)
+        focal = torch.rand(4, device=device, dtype=dtype)
+
+        points3d = kornia.geometry.depth.depth_from_disparity(disparity, baseline, focal)
+        assert points3d.shape == (batch_size, 1, 3, 4)
+
+    @pytest.mark.parametrize("shape", [1, 4, (3, 4)])
+    def test_tensors_for_baseline_focal(self, shape, device, dtype):
+        disparity = torch.rand(1, 1, 3, 4, device=device, dtype=dtype)
+        baseline = torch.rand(shape, device=device, dtype=dtype)
+        focal = torch.rand(shape, device=device, dtype=dtype)
+
+        points3d = kornia.geometry.depth.depth_from_disparity(disparity, baseline, focal)
+        assert points3d.shape == (1, 1, 3, 4)
+
+    def test_simple(self, device, dtype):
+        # this is for default normalize_points=False
+        disparity = 2 * torch.tensor(
+            [[[[1.0, 1.0, 1.0], [1.0, 1.0, 1.0], [1.0, 1.0, 1.0], [1.0, 1.0, 1.0]]]], device=device, dtype=dtype
+        )
+
+        baseline = torch.tensor([1.0, 0.0, 0.0], device=device, dtype=dtype)
+        focal = torch.tensor([1.0, 0.0, 0.0], device=device, dtype=dtype)
+
+        depth_expected = torch.tensor(
+            [
+                [
+                    [
+                        [0.5000, 0.0000, 0.0000],
+                        [0.5000, 0.0000, 0.0000],
+                        [0.5000, 0.0000, 0.0000],
+                        [0.5000, 0.0000, 0.0000],
+                    ]
+                ]
+            ],
+            device=device,
+            dtype=dtype,
+        )
+
+        depth = kornia.geometry.depth.depth_from_disparity(disparity, baseline, focal)
+        assert_close(depth, depth_expected, rtol=1e-3, atol=1e-3)
+
+    def test_gradcheck(self, device, dtype):
```

Comment:
```suggestion
    def test_gradcheck(self, device):
```
In theory grad check is valid only for double precision. Tensor to grad check already converts